### PR TITLE
Define the list of ordered and valid channels.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: csclient/params/params.go [hunk 1]
```diff
@@ -58,6 +58,25 @@ const (
...
 
+// OrderedChannels holds the list of valid channels in order of publishing
+// status, most stable first.
+var OrderedChannels = []Channel{
+	StableChannel,
+	CandidateChannel,
+	BetaChannel,
+	EdgeChannel,
+	UnpublishedChannel,
+}
```
Comment (urosj) [inline csclient/params/params.go]: Description of "desired" channel semantics maybe?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/273/
Test PASSed.

**Comment by urosj** [comment]
+1 with a comment

File: csclient/params/params.go (hunk 1 above)
Comment (frankban) [inline csclient/params/params.go]: It's not just desired, this order is effectively used in the charm store.

**Comment by frankban** [comment]
Thanks for the review!

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**